x-pack/plugins/security_solution/public/detections/containers/detection_engine/rules/use_rule_from_timeline.test.ts

```diff
@@ -116,9 +117,9 @@ describe('useRuleFromTimeline', () => {
     });
 
     it('does not reset timeline sourcerer if it originally had same data view as the timeline used in the rule', async () => {
-      const { result, waitForNextUpdate } = renderHook(() => useRuleFromTimeline(setRuleQuery));
+      const { result } = renderHook(() => useRuleFromTimeline(setRuleQuery));
       expect(result.current.loading).toEqual(true);
-      await waitForNextUpdate();
+      await waitFor(() => null);
       expect(setRuleQuery).toHaveBeenCalled();
       expect(mockDispatch).toHaveBeenCalledTimes(2);
```

Comment:
> Not sure I'm getting what `await waitFor(() => null);` does.

Please refer to the PR description for the implication of `await waitFor(() => null)`,  

> And given that this test is failing locally, I believe the correct way of writing this would be by placing assertions inside of `waitFor`

I do agree that placing the assertion within the `waitFor` callback would be correct, the choice to replace `waitForNextUpdate` simply with `await waitFor(() => null)` is a tradeoff that was made between having a much more lengthy PR and getting this implemented with the least amount of change, this rationale is also outlined in the PR description... regardless thanks for pointing this out.